jax/dtypes.py

```diff
@@ -116,6 +117,8 @@ def coerce_to_array(x):
   rules.
   """
   dtype = python_scalar_dtypes.get(type(x), None)
+  if np.any(x >= (2 ** 31)) and dtype in [onp.dtype('int32'), onp.dtype('uint32'), onp.dtype('float32')]:
```

Comment:
This should only include `int32` and `uint32` (and should be specific to the ranges of those two dtypes, which you can get through the `min` and `max` properties of `onp.iinfo(onp.int32)` and `onp.iinfo(onp.uint32)`). `float32` has a much larger range.

You also only want to run this check on Python scalars, since numpy arrays are already guaranteed to fit in the limits imposed by their dtype.

Reply:
thanks @jekbradbury !